spot-electron/src/volume-control/volume-control.js

```diff
@@ -0,0 +1,130 @@
+const _ = require('lodash');
+const os = require('os');
+const osascript = require('node-osascript');
+
+const { clientController } = require('../client-control');
+const { logger } = require('../logger');
+
+const VOLUME_STEP = 5;
+
+/**
+ * Implements a system volume control routine that can control the volume of several host OSes.
+ */
+class VolumeControl {
+    _platform;
+
+    /**
+     * Instantiates a new control.
+     *
+     * @param {string} platform - One of the values node.os.platform() may return.
+     */
+    constructor(platform) {
+        this._platform = _.capitalize(platform);
+    }
+
+    /**
+     * Decreases the volume by {@code VOLUME_STEP} percent.
+     *
+     * @returns {void}
+     */
+    decreaseVolume() {
+        this._adjustVolume(VOLUME_STEP * -1);
+    }
+
+    /**
+     * Increases the volume by {@code VOLUME_STEP} percent.
+     *
+     * @returns {void}
+     */
+    increaseVolume() {
+        this._adjustVolume(VOLUME_STEP);
+    }
+
+    /**
+     * Adjusts the volume by a percent value.
+     *
+     * @param {number} percentage - The value.
+     * @returns {void}
+     */
+    _adjustVolume(percentage) {
+        this._getVolume().then(value => {
+            const newVolume = Math.min(100, value * (100 + percentage) / 100);
```

Comment:
If value is ever zero then I think volume would never decrease,

Reply:
Right... I think I was drunk when I wrote this

Reply:
I meant it would never increase in volume, not never decrease. I must have had sympathy drunkenness.